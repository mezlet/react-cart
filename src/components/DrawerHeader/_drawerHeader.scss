
   .drawer-left-icon {
      background: rgba(97, 100, 91, 0.151) !important;
      border-radius: 50%;
      padding: .1rem;
      height: 1.3rem;
      width: 1.3rem;
      display: flex;
      justify-content: center;
      align-items: center;
      font-size: 23px;
      cursor: pointer !important;
  }
   