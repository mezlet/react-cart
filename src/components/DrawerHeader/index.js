import React from 'react';
import PropTypes from 'prop-types';

const DrawerHeader = ({ toggleDrawer }) => {
	return (
		<div className='d-flex my-3'>
			<div className='drawer-left-icon' onClick={toggleDrawer} data-test='close-drawer-btn'>
				&#8249;
			</div>
			<h5 className='mx-3'>Your Cart</h5>
		</div>
	);
};

DrawerHeader.propTypes = {
	toggleDrawer: PropTypes.func.isRequired
}
export default DrawerHeader;
