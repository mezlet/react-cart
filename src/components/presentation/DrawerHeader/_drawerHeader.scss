
   .drawer-icon {
      background: rgba(97, 100, 91, 0.048);
      border-radius: 50%;
      padding: .1rem;
      height: 1.3rem;
      width: 1.3rem;
      display: flex;
      justify-content: center;
      align-items: center;
      font-size: 28px;
      cursor: pointer;
   }