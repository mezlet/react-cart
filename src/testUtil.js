import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
    return checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
}