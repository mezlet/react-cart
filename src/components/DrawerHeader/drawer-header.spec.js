import React from 'react';
import { shallow } from 'enzyme';
import DrawerHeader from './index';
import { checkProps} from '../../testUtil';

const props = {
    toggleDrawer: jest.fn()
}

const setUp = (props = {}) => {
    const component = shallow(<DrawerHeader {...props} />);
    return component;
}

describe('DrawerHeader', () => {
    let component;
    beforeEach(() => {
        component = setUp(props);
    })
    it('should render without errors', () => {
        expect(component).toMatchSnapshot(); 
    });
    it('should render close drawer button', () => {
        const wrapper = component.find(`[data-test='close-drawer-btn']`);
        expect(wrapper.length).toBe(1); 
    })
    it('Checking PropTypes should not throw error', () => {
        const propErr = checkProps(DrawerHeader, props);
        expect(propErr).toBeUndefined();
    })
    it('should toggle drawer', () => {
        const button = component.find(`[data-test='close-drawer-btn']`);
        button.simulate('click');
        expect(props.toggleDrawer.mock.calls.length).toBe(1);
    })
})