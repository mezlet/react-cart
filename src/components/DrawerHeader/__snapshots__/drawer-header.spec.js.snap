// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader should render without errors 1`] = `ShallowWrapper {}`;
